crates/solver/src/settlement_rater.rs

```diff
@@ -235,14 +264,34 @@ impl SettlementRating for SettlementRater {
                 },
             )| {
                 match gas_estimate {
-                    Ok(gas_estimate) => Either::Left((
-                        simulation.solver,
-                        rate_settlement(i, simulation.settlement, gas_estimate),
-                        simulation.transaction.access_list,
-                    )),
+                    Ok(gas_estimate) => {
+                        let gas_limit = gas_limit_for_estimate(gas_estimate);
+                        let required_balance = gas_limit
+                            .saturating_mul(U256::from_f64_lossy(gas_price.max_fee_per_gas));
+                        let solver_balance = solver_balances
+                            .get(&simulation.solver.account().address())
+                            .copied()
+                            .unwrap_or_default();
```

Comment:
I think more appropriate would be to return a separate error in case the web3 call to get the balance failed. Otherwise solvers might get confused then they receive `SimulationError::InsufficientBalance` when they actually had enough funds.
But I expect this to be rare enough that it might not be worth it to make the code even more noisy.

Reply:
> But I expect this to be rare enough that it might not be worth it to make the code even more noisy.

This was my rationale. Also, we are moving to co-location soon so I don't want to optimize too much for solver UX here.